packages/kit/src/runtime/server/page/index.js

```diff
@@ -18,24 +18,21 @@ import { get_data_json } from '../data/index.js';
 
 /**
  * @param {import('types').RequestEvent} event
- * @param {import('types').SSRRoute} route
  * @param {import('types').PageNodeIndexes} page
  * @param {import('types').SSROptions} options
  * @param {import('types').SSRManifest} manifest
  * @param {import('types').SSRState} state
  * @param {import('types').RequiredResolveOptions} resolve_opts
  * @returns {Promise<Response>}
  */
-export async function render_page(event, route, page, options, manifest, state, resolve_opts) {
-	if (state.initiator === route) {
+export async function render_page(event, page, options, manifest, state, resolve_opts) {
+	if (state.initiator === event.request.url) {
```

Comment:
This now compares by url, not matched route anymore. Does this have any buggy implications that I'm not aware of?

That infinite loop check is a bit too simple anyway, if you have a loop that spans more than one fetch we don't catch it right now - but it also doesn't seem to have happened to anyone yet, so something to worry about another day I'd say.

Reply:
Not opposed to this change but what's the reason for it?

Reply:
ah nvm, i see the explanation in the original comment. i honestly don't know if it's potentially buggy

Reply:
Also this doesn't catch redirects right? I'm wondering at this point if this just is some ancient guard we no longer need.

Reply:
We do need it, otherwise there's no way to fall back to `fetch` in the case where `event.fetch` is used to request something other than a route in the currently available manifest, no way to bail out in the case where the root layout fetches a non-existent resource, and no way to bail out when a route fetches itself. If we get rid of it a bunch of tests start failing.

Thinking further, using the URL rather than the `route` object does make me nervous. It fails in these edge cases, resulting in an infinite loop:

* `${url.pathname}?random=${Math.random()}` (would be fixed by using `event.url.pathname`)
* `${url.pathname}x` when matched against `/[...catchall]` (would not)

These might be vanishingly unlikely but IIRC the `initiator` stuff was added because we encountered a similarly vanishingly unlikely scenario in the wild. Infinite request loops can result in huge compute bills (I've seen this happen!) so we should strenuously avoid them where we can